pandas/core/tools/datetimes.py

```diff
@@ -316,8 +317,21 @@ def _convert_listlike_datetimes(
     elif unit is not None:
         if format is not None:
             raise ValueError("cannot specify both format and unit")
-        arg = getattr(arg, "values", arg)
-        result, tz_parsed = tslib.array_with_unit_to_datetime(arg, unit, errors=errors)
+        arg = getattr(arg, "_values", arg)
+
+        # GH 30050 pass an ndarray to tslib.array_with_unit_to_datetime
+        # because it expects an ndarray argument
+        if isinstance(arg, IntegerArray):
+            # Explicitly pass NaT mask to array_with_unit_to_datetime
+            mask = arg.isna()
+            arg = getattr(arg, "_ndarray_values", arg)
```

Comment:
Why getattr instead of `arg._ndarray_values`?

Reply:
Once we have IntegerArray.to_numpy, I think this could be something like `arg.to_numpy("i8", fill_value=NPY_NAT)`.

Reply:
yes would like to do this exactly (or a way to get data/mask publicly would be ok here as well). but what you are suggesting is nice.